From bb78992a3b614cfd3306f94ba7cc615f2a70bfba Mon Sep 17 00:00:00 2001
From: Meng Mingming <mingming.meng@nxp.com>
Date: Fri, 28 Oct 2016 11:18:40 +0800
Subject: [PATCH 2/2] MGS-2343-1 [#ccc] XWLD: T3DStressTest_Wayland displays
 abnormally while using G2D compositor

Move g2d_finish to other place for performance reason.

Upstream-Status: Inappropriate [i.MX specific]

Date: Oct 27, 2016
Signed-off-by: Meng Mingming <mingming.meng@nxp.com>
---
 src/g2d-renderer.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/g2d-renderer.c b/src/g2d-renderer.c
index 89c69bf..41c4d9e 100644
--- a/src/g2d-renderer.c
+++ b/src/g2d-renderer.c
@@ -609,7 +609,6 @@ repaint_region(struct weston_view *ev, struct weston_output *output, struct g2d_
 			g2d_set_clipping(gr->handle, clipRect.left, clipRect.top, clipRect.right, clipRect.bottom);
 			g2d_blitSurface(gr->handle, &gs->g2d_surface, dstsurface, &srcRect, &dstrect);
 		}
-		g2d_finish(gr->handle);
 	}
 }
 
@@ -680,6 +679,8 @@ g2d_renderer_repaint_output(struct weston_output *output,
 			     pixman_region32_t *output_damage)
 {
 	struct g2d_output_state *go = get_output_state(output);
+	struct weston_compositor *compositor = output->compositor;
+	struct g2d_renderer *gr = get_renderer(compositor);
 	int i;
 
 	use_output(output);
@@ -692,6 +693,7 @@ g2d_renderer_repaint_output(struct weston_output *output,
 			      &go->buffer_damage[go->current_buffer]);
 
 	repaint_views(output, output_damage);
+	g2d_finish(gr->handle);
 
 	pixman_region32_copy(&output->previous_damage, output_damage);
 	wl_signal_emit(&output->frame_signal, output);
-- 
2.7.4

